compat/src/test/scala/scala/util/ChainingOpsTest.scala

```diff
@@ -0,0 +1,37 @@
+package scala.util
+
+import org.junit.Assert._
+import org.junit.Test
+
+//import scala.tools.reflect.ToolBoxError
+//import scala.tools.testkit.RunTesting
+
+class ChainingOpsTest {
+  import scala.util.chaining._
+
+  @Test
+  def testAnyTap(): Unit = {
+    var x: Int = 0
+    val result = List(1, 2, 3)
+      .tap(xs => x = xs.head)
+
+    assertEquals(1, x)
+    assertEquals(List(1, 2, 3), result)
+  }
+
+  @Test def testAnyValTap(): Unit = assertEquals(42.tap(x => x), 42)
+
+  @Test
+  def testAnyPipe(): Unit = {
+    val times6 = (_: Int) * 6
+    val result = (1 - 2 - 3)
+      .pipe(times6)
+      .pipe(scala.math.abs)
+
+    assertEquals(24, result)
+  }
+
+//  @Test(expected = classOf[ToolBoxError])
+//  def testNoSelf(): Unit =
+//    runner.run("import scala.util.chaining._; Nil.self")
```

Comment:
not sure if there's a good way to keep this test

Reply:
We need to depend on `scala.tools.testkit.RunTesting`? Is this thing published?

Reply:
it is, apparently (https://repo1.maven.org/maven2/org/scala-lang/scala-testkit/2.13.1/). but it's only published for 2.13.1, which rather defeats the purpose :(

Reply:
It's also considered internal, the API could break at any time.